ecommerce/extensions/payment/processors/paystack.py

```diff
@@ -0,0 +1,199 @@
+            a payment from being created.
+        """
+        data = {
+            'amount': self.get_basket_amount(basket),
+            'email': basket.owner.email,
+            'callback_url': self.return_url,
+            'metadata': {
+                'cancel_action:': self.cancel_url,
+                'order_number': basket.order_number,
+                'basket_id': basket.id,
+                'custom_fields': self.get_paystack_custom_fields_data(basket)
+            },
+        }
+
+        success, response = self.paystack_client.handler(INITIALIZE_TRANSACTION_CODE, data)
+
+        if success:
+            logger.info("Successfully got hosted Paystack payment page for basket [%d].", basket.id)
+            data = response.get('data')
+            if data:
+                return {'payment_page_url': data.get('authorization_url')}
+
+        logger.error("Failed to get Paystack payment form for basket [%d].", basket.id)
+        raise GatewayError("Paystack payment creation failure: unable to get Paystack form token")
+
+    def handle_processor_response(self, response, basket=None):
+        """
+        Execute an approved Paystack transaction. This method will record payment processor
+        response for future usage.
+
+        Arguments:
+            response: Transaction response received from Paystack after successfull payment
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Returns:
+            HandledProcessorResponse
+        """
+        authorization = response.get('authorization')
+        transaction_id = response.get('id')
+
+        self.record_processor_response(authorization, transaction_id=transaction_id, basket=basket)
+        logger.info("Successfully executed Paystack payment [%s] for basket [%d].", transaction_id, basket.id)
+
+        currency = response.get('currency')
+        total = float(response.get('amount'))
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=authorization.get('last4'),
+            card_type=authorization.get('card_type')
+        )
+
+    def verify_refund(self, reference_number):
+        """
+        Returns a boolean value indicating refund object status after verification from Paystack
+        """
+        success, response = self.paystack_client.handler(FETCH_REFUND_CODE, reference_number)
+        if success:
+            data = response.get('data')[0]
+            refund_status = data.get('status')
+            logger.info("Paystack refund has been fetched with status: %s ", refund_status)
+
+            if refund_status == 'processed':
+                return True
+
+        logger.error("Unable to fetch refund object from paystack for refund_id: %s", reference_number)
+        return False
+
+    def issue_credit(self, order_number, basket, reference_number, amount, currency):
```

Comment:
can we rename this method name e.g `refund_credit_to_leaner`, ignore if this is an overridden method.

Reply:
overridden method.